kernel/Kconfig

```diff
@@ -600,6 +600,15 @@ config SYSTEM_WORKQUEUE_NO_YIELD
 	  cooperative and a sequence of work items is expected to complete
 	  without yielding.
 
+config SYSTEM_WORKQUEUE_BLOCKING
+	bool "Select whether system work queue allows blocking calls"
```

Comment:
The `default n` of SYSTEM_WORKQUEUE_BLOCKING is a change in behavior and may break existing applications. These applications may or may not have a risk of running into deadlocks.

Therefore, I believe we should consider setting the default to `y` and issue a warning for a deprecation period.





. Should there be a deprecation period where there is initially

Reply:
We can add `default y`, and deprecate it immediately, I'm all for that :)

Reply:
> We can add `default y`, and deprecate it immediately, I'm all for that :)

Wouldn't that go against policy? i.e. no use of deprecated symbols in the Zephyr tree, so if set to `y` by default and it's deprecated, then it needs to be set to `=n` in all tests, samples, etc.?

Reply:
Yeah, I immediately noticed this would require disabling it everywhere manually :) not optimal